shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz_hints/presentation/StepQuizHintsActionDispatcher.kt

```diff
@@ -37,19 +37,26 @@ class StepQuizHintsActionDispatcher(
                 val hintsIds = stepQuizHintsInteractor.getNotSeenHintsIds(action.stepId)
                 val dailyStepId = profileInteractor
                     .getCurrentProfile()
-                    .fold(
-                        onSuccess = { it.dailyStep },
-                        onFailure = { null }
-                    )
+                    .map { it.dailyStep }
+                    .getOrNull()
 
                 val lastSeenHint = stepQuizHintsInteractor.getLastSeenHint(action.stepId)
 
+                var lastSeenHintHasReaction = false
+                if (lastSeenHint != null) {
+                    val hintState = stepQuizHintsInteractor
+                        .getCachedHintState(action.stepId, lastSeenHint.id)
+                        .getOrNull()
+                    lastSeenHintHasReaction = hintState?.hasReaction ?: false
+                }
```

Comment:
It's better to not use `var` variables if possible.
```suggestion
                val lastSeenHintHasReaction = stepQuizHintsInteractor
                        .getCachedHintState(action.stepId, lastSeenHint.id)
                        .getOrNull()?.hasReaction ?: false
```

Reply:
I think your code suggestion will not compile because `lastSeenHint` is optional and we need to safely unwrap it before.

Reply:
 Ok, then you can declare `lastSeenHintHasReaction` as a result of `if/else` expression.
 `val lastSeenHintHasReaction = if (lastSeenHint != null) {} else { false }`